### ISSUE TITLE
simpleMessageListenerContainer.stop(...) takes very long time ~20 sec

### ISSUE THREAD

**Issue by bstoilov** [spring-cloud/spring-cloud-aws#543]
This is more of a question than a bug.
I have SQS listeners and for some tests I need to stop the listener, but it takes very long time to stop the queue. more than 20 sec. I even had to increase the timeout.

in simpleMessageListenerContainer.stop method

```
Future<?> future = this.scheduledFutureByQueue.remove(logicalQueueName);
  if (future != null) {
    future.get(this.queueStopTimeout, TimeUnit.MILLISECONDS);
}
```

the future is the thing that takes  20 sec.

**Comment by maciejwalkowiak** [comment]
For the test purposes you can overwrite `SimpleMessageListenerContainerFactory` bean to stop immediately:

```
@Bean
SimpleMessageListenerContainerFactory simpleMessageListenerContainerFactory() {
	SimpleMessageListenerContainerFactory factory = new SimpleMessageListenerContainerFactory();
	factory.setWaitTimeOut(0);
	return factory;
}
```

Does it answer your question?

**Comment by maciejwalkowiak** [comment]
`SimpleListenerContainerFactory` has now a property `queueStopTimeout` that you can use to control this behaviour (https://github.com/spring-cloud/spring-cloud-aws/commit/e11c184c9d798a1c0425e296c186d291a46708d4). It will be released in 2.3.
